File: tests/OsloveniTest.js
// This unit test is written in Node.js and requires module unit.js installed.
const test = require('unit.js');
const tools = require('../javascript/osloveni');
const fs = require('fs');

fs.readFile('../generate-code/words.txt', 'utf8', (err, data) => {
  test.value(err).isNull()
  const lineSplit = data.split(/\r?\n/);
  lineSplit.forEach((line) => {
      if (line.charCodeAt(0) == 65279)
        line = line.substring(1)  // strip UTF-8 file header
      const wordPair = line.split(';')
      if (wordPair[0] != '') {
          test.value(wordPair[1].toLowerCase()).is(tools.osloveni(wordPair[0].toLowerCase())) // lowercase
          test.value(wordPair[1].toUpperCase()).is(tools.osloveni(wordPair[0]).toUpperCase()) // UPPERCASE
          test.value(wordPair[1].replace(/\w\S*/g, function(txt){{return txt.charAt(0).toUpperCase() + txt.substring(1).toLowerCase();}})).is(tools.osloveni(wordPair[0]).replace(/\w\S*/g, function(txt){{return txt.charAt(0).toUpperCase() + txt.substring(1).toLowerCase();}})) // TitleCase
      }
  })
});

File: tests/OsloveniTest.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/UnitTests/JUnit5TestClass.java to edit this template
 */

import static Osloveni.osloveni;
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Paths;
import java.util.logging.Level;
import java.util.logging.Logger;
import org.junit.jupiter.api.AfterEach;
import org.junit.jupiter.api.AfterAll;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

/**
 *
 * @author karlos
 */
public class OsloveniTest {
    private static final Logger logger = Logger.getLogger(OsloveniTest.class.getName());
    public OsloveniTest() {
    }

    @org.junit.jupiter.api.BeforeAll
    public static void setUpClass() throws Exception {
    }

    @org.junit.jupiter.api.AfterAll
    public static void tearDownClass() throws Exception {
    }

    @org.junit.jupiter.api.BeforeEach
    public void setUp() throws Exception {
    }

    @org.junit.jupiter.api.AfterEach
    public void tearDown() throws Exception {
    }
    
    /**
     * Test of osloveni method, of class Osloveni.
     */
    @org.junit.jupiter.api.Test
    public void testOsloveni() {
        System.out.println("osloveni");
        
        try {
            String data = new String(
                    Files.readAllBytes(Paths.get("../generate-code/words.txt")));
            String[] lines = data.split("\r\n");
            for (String line : lines) {
                String[] words = line.split(";");
                String jmeno1 = words[0];
                String jmeno5 = words[1];
                String result = osloveni(jmeno1);
                assertEquals(jmeno5, result);

                jmeno1 = words[0].toLowerCase();
                jmeno5 = words[1].toLowerCase();
                result = osloveni(jmeno1);
                assertEquals(jmeno5, result);

                jmeno1 = words[0].toUpperCase();
                jmeno5 = words[1].toUpperCase();
                result = osloveni(jmeno1);
                assertEquals(jmeno5, result);
            }
        } catch (IOException ex) {
            Logger.getLogger(Osloveni.class.getName()).log(Level.SEVERE, null, ex);
        }
        
        
    }

    /**
     * Test of capitalize method, of class Osloveni.
     */
    @org.junit.jupiter.api.Test
    public void testCapitalize() {
        System.out.println("capitalize");
        String text = "hello";
        String expResult = "Hello";
        String result = Osloveni.capitalize(text);
        assertEquals(expResult, result);
    }
    
}
